src/qibolab/result.py

```diff
@@ -61,8 +32,8 @@ def q(self):
         return self.array.q
 
     def __add__(self, data):
-        i = np.append(self.i, data.i, axis=0)
-        q = np.append(self.q, data.q, axis=0)
+        i = np.append(self.i, data.i)
+        q = np.append(self.q, data.q)
```

Comment:
Why are you dropping the axis argument? If you do it, you would force flattening of the arrays, loosing shape information

> **axis: int, optional**
>    The axis along which values are appended. If axis is not given, both arr and values are flattened before use.

https://numpy.org/doc/stable/reference/generated/numpy.append.htm

Reply:
Currently if I keep tests will not pass (not sure why). I can investigate and find a solution.

Reply:
Fixed. The problem was that in the `average` property I was computing the mean using `.mean()` which was changing `i` and `q` from array to float. I solved the issue using `np.mean(.., keepdims=True)`.

Reply:
I believe the solution to be good enough for the time being, and definitely was a subtle issue.

But (I believe) the optimal one would be to just add one dimension here, if needed.
Or, even better, `np.append` is doing on its own, so let's just do the following:
```py
axis = 0 if len(self.i) > 0 else None
i = np.append(self.i, data.i, axis=axis)
...
```

In this way you can drop `keepdims`, and we preserve a meaningful tensor rank after averaging.

Reply:
I understand what you are trying to do.
Actually this solution doesn't work since it generates this error `TypeError: len() of unsized object`
```py
import numpy as np
from qibolab.result import ExecutionResults
i = np.random.rand(10)
q = np.random.rand(10)
test = ExecutionResults.from_components(i, q)
len(test.average.array.i)
```
A possible alternative would to use shape in order to decide when to add one dimension.
However, also this solution fails if you try to add two `AveragedResults` it will fail with `ValueError: zero-dimensional arrays cannot be concatenated`.

So I think that if we don't want to over complicate things we could just leave `keepdims` for now.
Let me know if you agree.

Reply:
Thanks now it works!